Fix auto imports and go to sources for worksheets outside sources

tgodzik opened the pull request:
The sources and classpath was not forwarded to the symbol search for worksheets outside build targets.

Fixes https://github.com/scalameta/metals/issues/2029

kpbochenek reviewed (approved):
I think with this test change we stop testing completion for worksheets inside build target?
btw. this test tests completions, there is a test in line ~400 called "definition"

Maybe we could simply have 3 dedicated tests:

- gotodef-dependency-inside-target         <- this should already work
- gotodef-dependency-outside-target       <- this PR fixes this
- gotodef-targetdependency                      <- inside target we can also use project deps, correct?

This is something 'nice to have' and you can completely ignore it :)

tgodzik commented:
> I think with this test change we stop testing completion for worksheets inside build target?
> btw. this test tests completions, there is a test in line ~400 called "definition"

This test deals with the worksheet dependencies and I am avoiding adding additional tests and we still test completions in other tests. 
 
> * gotodef-dependency-inside-target         <- this should already work

I can add it back to the test, no problem.

> * gotodef-targetdependency                      <- inside target we can also use project deps, correct?

Already tested in another test.

tgodzik commented:
Moved the test to WorksheetLspSuite, which is tested for Scala 2.12.12 and that test was anyway only tested for that version. Also, split it into two tests one for outside target and one for inside it.

kpbochenek reviewed (approved):
I start to appreciate elasticity of munit more and more :D

LGTM :+1:
